1_19_R2/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_19_R2.java

```diff
@@ -0,0 +1,1565 @@
+        final Object nmsValue;
+
+        if (value instanceof Location l) {
+            if (key.equals("nearest_bed") || key.equals("celebrate_location") || key.equals("nearest_repellent")) nmsValue = new BlockPos(l.getX(), l.getY(), l.getZ());
+            else nmsValue = GlobalPos.of(toNMS(l.getWorld()).dimension(), new BlockPos(l.getX(), l.getY(), l.getZ()));
+        }
+        else if (value instanceof Location[] ls) {
+            List<GlobalPos> p = new ArrayList<>();
+
+            for (Location l : ls) {
+                p.add(GlobalPos.of(toNMS(l.getWorld()).dimension(), new BlockPos(l.getX(), l.getY(), l.getZ())));
+            }
+
+            nmsValue = p;
+        }
+        else if (value instanceof Player p) {
+            if (key.equals("liked_player")) nmsValue = p.getUniqueId();
+            else nmsValue = toNMS(p);
+        }
+        else if (value instanceof Memory.WalkingTarget t) nmsValue = new WalkTarget(toNMS(t.getLocation()), (float) t.getSpeedModifier(), t.getDistance());
+        else if (value instanceof LivingEntity l) nmsValue = toNMS(l);
+        else if (value instanceof Entity e) {
+            if (key.equals("angry_at")) nmsValue = e.getUniqueId();
+            else nmsValue = toNMS(e);
+        }
+        else if (value instanceof org.bukkit.block.Block[] b) {
+            final Collection<GlobalPos> s;
+            if (key.equals("doors_to_close")) s = new HashSet<>();
+            else s = new ArrayList<>();
+
+            for (org.bukkit.block.Block bl : b) {
+                Location l = bl.getLocation();
+                s.add(GlobalPos.of(toNMS(l.getWorld()).dimension(), new BlockPos(l.getX(), l.getY(), l.getZ())));
+            }
+            nmsValue = s;
+        }
+        else if (value instanceof Villager[] vs) {
+            List<net.minecraft.world.entity.LivingEntity> s = new ArrayList<>();
+            for (Villager v : vs) s.add(toNMS(v));
+            nmsValue = s;
+        }
+        else if (value instanceof Player[] ps) {
+            List<net.minecraft.world.entity.player.Player> s = new ArrayList<>();
+            for (Player p : ps) s.add(toNMS(p));
+            nmsValue = s;
+        }
+        else if (value instanceof LivingEntity[] ls) {
+            List<net.minecraft.world.entity.LivingEntity> s = new ArrayList<>();
+            for (LivingEntity l : ls) s.add(toNMS(l));
+            nmsValue = s;
+        }
+        else if (value instanceof EntityDamageEvent.DamageCause c) nmsValue = toNMS(c);
+        else nmsValue = value;
+
+        return nmsValue;
+    }
+
+    public static Object fromNMS(Mob m, String key, Object nmsValue) {
+        Object value = nmsValue;
+
+        if (nmsValue instanceof GlobalPos l) {
+            BlockPos pos = l.pos();
+            World w = ((CraftServer) Bukkit.getServer()).getHandle().getServer().registries().compositeAccess().registryOrThrow(Registries.DIMENSION).get(l.dimension()).getWorld();
```

Comment:
The field `MinecraftServer#registryHolder` was replaced by `MinecraftServer#registries()`, but `registries()` returns a `LayeredRegistryAccess`. I've replaced it with `registries().compositeAccess()` but I don't know if that does the same thing or not.

Reply:
It seems to return a type of "Frozen" registry, but that might only be affected when it tries writing things. Seems to work.
I'll test this when I get home.

Reply:
Seems like my original thought was correct: writing isn't allowed and requires hacky things to happen, but reading is perfectly ok, which is what we're trying to do.